### PR TITLE
Add auth bypass id to whitehall endpoint

### PR DESCRIPTION
**Pull request by BeckaL**
Some foundational work to allow shareable previews for Whitehall to access assets - Whitehall has its own create endpoint, so this PR adds auth_bypass_id (a UUID that allows for non-sign-on access to an edition and its related content via a token) to the allowed parameters for Whitehall assets. This is identical to what is done for non-Whitehall assets.

This commit also introduces some tests for the WhitehallMediaController to ensure that an asset can be accessed with an access token containing an auth_bypass_id. This repeats some tests in the non-Whitehall media controller, and the behaviour is already defined for the WhitehallMediaController via inheritance, but I thought the tests were useful from a documentation standpoint.

### CODE REVIEW

File: spec/controllers/whitehall_media_controller_spec.rb [hunk 1]
```diff
@@ -205,6 +205,84 @@
...
+        end
+      end
+
+      context "when a user is not authenticated and has provided an invalid token" do
+        before { not_logged_in }
+
+        it "authenticates the user" do
+          cookies["auth_bypass_token"] = "bad-token"
+          expect(controller).to receive(:authenticate_user!)
+          get :download, params: { path: path, format: format }
```
Comment (brucebolt) [inline spec/controllers/whitehall_media_controller_spec.rb:272]: Is there an expectation here (e.g. that the response is unsuccessful) that could be added?

**Review by brucebolt — COMMENTED** [review]
Looks good to me 👍 .

There's just one test that I think is missing the assertion.

File: spec/controllers/whitehall_media_controller_spec.rb (hunk 1 above)
Comment (BeckaL) [reply spec/controllers/whitehall_media_controller_spec.rb:272]: Just writing this here for ref - as discussed, have added some requests specs for whitehall that cover this scenario.

**Review by brucebolt — APPROVED** [review]
Looks good to me 👍 .

Thanks for persevering with the tests!

**Pull request merged**